app/move/steps/update.js

```diff
@@ -1,44 +1,88 @@
-const { cloneDeep } = require('lodash')
-
 const {
   Assessment,
   MoveDate,
-  MoveDetails,
+  // TODO: reenable when redirect api available
+  // MoveDetails,
   PersonalDetails,
 } = require('../controllers/update')
 
 const createSteps = require('./create')
 
-const updateControllers = {
-  '/personal-details': PersonalDetails,
-  '/move-details': MoveDetails,
-  '/move-date': MoveDate,
-  '/court-information': Assessment,
-  '/risk-information': Assessment,
-  '/health-information': Assessment,
-}
-
-const provideStepProps = (key, step) => {
-  const stepProps = {
-    entryPoint: true,
-    backLink: null,
-    next: undefined,
-    buttonText: 'actions::save_and_continue',
-  }
-
-  const updatedStep = {
-    ...cloneDeep(step),
-    ...stepProps,
-  }
-
-  if (updateControllers[key]) {
-    updatedStep.controller = updateControllers[key]
-  }
-  return updatedStep
+const updateStepPropOverrides = {
+  entryPoint: true,
+  backLink: null,
+  next: undefined,
+  buttonText: 'actions::save_and_continue',
 }
 
-const updateSteps = Object.keys(createSteps)
-  .filter(key => updateControllers[key])
-  .map(key => ({ [key]: provideStepProps(key, createSteps[key]) }))
+const updateSteps = [
+  {
+    key: 'personal_details',
+    role: 'move:update',
```

Comment:
I wonder if this is the wrong term. We've tried really hard to not have "roles" within the app.

When a user signs in, with given "roles", we assign permissions, so maybe if that's what we're trying to do here we should use `permissions` as an array for consistency?

Reply:
`permission` fine. Why an array though? This gets passed to `protectRoute` which takes a string

Reply:
Ah yes. That middleware only support a string. And if we need to protect based on multiple permissions we have used that middleware twice.

Maybe you need to cater for that here though. A user may require `move:update` as well as `move:update:documents` for a particular step. Passing just `move:update:documents` wouldn't feel enough as we would require that user to have the permission to update first as well as update at a granular level.

Reply:
Is this to support STC/SCH where only they can edit/upload documents?

At the moment, that logic is actually tied to the `from_location` rather than a specific permission because of the way it was originally required. We may want the ability to allow other establishments to upload documents, regardless of that user's roles/permissions.

If it's just to support documents at this stage, this may not be the best place for that logic if we need to keep it tied to `from_location`. That logic could live in the [documents controller](https://github.com/ministryofjustice/hmpps-book-secure-move-frontend/blob/95be55df32857b54a6a3193cba8c469b524ebf20/app/move/controllers/create/document.js#L16) instead.

Reply:
It also allows canAccess to be called within the controller using the value from the config